homeassistant/components/switchbot_cloud/climate.py

```diff
@@ -0,0 +1,118 @@
+"""Support for SwitchBot Air Conditioner remotes."""
+
+from typing import Any
+
+from switchbot_api import AirConditionerCommands
+
+import homeassistant.components.climate as FanState
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .entity import SwitchBotCloudEntity
+
+_SWITCHBOT_HVAC_MODES: dict[HVACMode, int] = {
+    HVACMode.HEAT_COOL: 1,
+    HVACMode.COOL: 2,
+    HVACMode.DRY: 3,
+    HVACMode.FAN_ONLY: 4,
+    HVACMode.HEAT: 5,
+}
+
+_DEFAULT_SWITCHBOT_HVAC_MODE = _SWITCHBOT_HVAC_MODES[HVACMode.FAN_ONLY]
+
+_SWITCHBOT_FAN_MODES: dict[str, int] = {
+    FanState.FAN_AUTO: 1,
+    FanState.FAN_LOW: 2,
+    FanState.FAN_MEDIUM: 3,
+    FanState.FAN_HIGH: 4,
+}
+
+_DEFAULT_SWITCHBOT_FAN_MODE = _SWITCHBOT_FAN_MODES[FanState.FAN_AUTO]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        SwitchBotCloudAirConditionner(data.api, device, coordinator)
+        for device, coordinator in data.devices.climates
+    )
+
+
+class SwitchBotCloudAirConditionner(SwitchBotCloudEntity, ClimateEntity):
+    """Representation of a SwitchBot air conditionner, as it is an IR device, we don't know the actual state."""
+
+    _attr_assumed_state = True
+    _attr_supported_features = (
+        ClimateEntityFeature.FAN_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_fan_modes = [
+        FanState.FAN_AUTO,
+        FanState.FAN_LOW,
+        FanState.FAN_MEDIUM,
+        FanState.FAN_HIGH,
+    ]
+    _attr_fan_mode = FanState.FAN_AUTO
+    _attr_hvac_modes = [
+        HVACMode.HEAT_COOL,
+        HVACMode.COOL,
+        HVACMode.DRY,
+        HVACMode.FAN_ONLY,
+        HVACMode.HEAT,
+    ]
+    _attr_hvac_mode = HVACMode.FAN_ONLY
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_target_temperature = 21
+    _attr_name = None
+
+    async def _do_send_command(
+        self,
+        hvac_mode: HVACMode | None = None,
+        fan_mode: str | None = None,
+        temperature: float | None = None,
+    ) -> None:
+        new_temperature = temperature or self._attr_target_temperature
```

Comment:
Do we support setting 0 degrees? In that case we need to check for `None` explicitly here.

Reply:
For Air Conditioner, it usually does not go bellow 15 degrees:
https://americanhomewater.com/how-many-degrees-can-an-air-conditioner-cool/
https://www.meridiancooling.co.uk/news/how-cold-can-air-conditioning-get

Reply:
Ok.